clang/lib/CConv/DeclRewriter.cpp

```diff
@@ -166,155 +166,179 @@ void DeclRewriter::rewriteParmVarDecl(ParmVarDeclReplacement *N) {
 }
 
 
-template <typename DT, DeclReplacement::DRKind DK>
-void DeclRewriter::rewriteMultiDecl(DeclReplacementTempl<DT, DK> *N,
-                                    RSet &ToRewrite) {
-  DT *D = N->getDecl();
-  std::string SRewrite = N->getReplacement();
-  if (Verbose) {
-    errs() << "Decl at:\n";
-    if (N->getStatement())
-      N->getStatement()->dump();
-  }
-  SourceRange TR = D->getSourceRange();
+template <typename DRType>
+void DeclRewriter::rewriteFieldOrVarDecl(DRType *N, RSet &ToRewrite) {
+  static_assert(std::is_same<DRType, FieldDeclReplacement>::value
+                    || std::is_same<DRType, VarDeclReplacement>::value,
+                "Method expects variable or field declaration replacement.");
 
-  // Is there an initializer? If there is, change TR so that it points
-  // to the START of the SourceRange of the initializer text, and drop
-  // an '=' token into sRewrite.
-  // Only Vardecls can have initializers
-  if(auto VD = dyn_cast<VarDecl>(D)) {
-    if (VD->hasInit()) {
-      SourceLocation EqLoc = VD->getInitializerStartLoc();
-      TR.setEnd(EqLoc);
-      SRewrite = SRewrite + " =";
-    } else {
-      // There is no initializer, lets add it.
-      if (isPointerType(VD) &&
-          (VD->getStorageClass() != StorageClass::SC_Extern))
-        SRewrite = SRewrite + " = ((void *)0)";
-      //MWH -- Solves issue 43. Should make it so we insert NULL if
-      // stdlib.h or stdlib_checked.h is included
-    }
+  if (isSingleDeclaration(N)) {
+    rewriteSingleDecl(N, ToRewrite);
+  } else if (Skip.find(N) == Skip.end()) {
```

Comment:
The `Skip` set is the set of names to skip during rewriting, I take it. Why does it only apply to multi-decls? I wonder if there's a better name for it (`Skip` is not so evocative for me) ?

Reply:
This is a bit of an odd point. The naming can presumably be improved. The reason `Skip` only contain declarations from multi-decls is that these are the only declarations that are at risk of being visited more than once. A single declaration is visited, rewritten, and then never seen again. A multi-decl will be rewritten when an earlier declaration in the same statement is visited, but it's seen again when the visitor eventually reaches it (or any other declaration in the same multi-decl statement).

Reply:
Your comment here is helpful; some form of it could go in the code itself.